modules/tf-cloud-credential/data.tf

```diff
@@ -1,4 +1,4 @@
-resource "tfe_workspace" "workspace" {
-  name         = var.workspace_name_prefix
+data "tfe_workspace" "workspace" {
```

Comment:
@JoseD92 This is partly my fault because I should have caught this when I was reviewing the original PR for this but this needs to be a data source and not a resource. 

The workspace is created when we run `tf init` on a project that has TF Cloud set as its remote backend. We therefore just need to use a datasource to look it up. Projects that will be using this module will not be in TF Cloud (since they are setting up variables to make using TF Cloud against AWS possible) so we should leave it up to the projects to create themselves and keep that step separate from this module.

Reply:
@JoseD92 Also I wouldn't recommend putting resources in a file named `data.tf`. That isn't a hard rule but it's probably best to use `data.tf` for datasources.

Reply:
@mcgirr is there an update to make here, or is the PR good to merge as-is?

Reply:
Nope, this is ready to merge as it. I was using PR comment for feedback for the earlier work that @JoseD92 did here.